src/index.js

```diff
@@ -206,28 +197,38 @@ function showMessage(text) {
 }
 
 function saveData() {
-    localStorage.setItem('info', JSON.stringify(currentPageInfo));
-    localStorage.setItem('category', category);
+  let toSave = JSON.stringify(currentPageInfo);
+  let saved = localStorage.getItem('info');
+    
+    if (saved === toSave) {
+        alert('Data is already saved!')
+    } else {
+        localStorage.setItem('info', toSave);
+        localStorage.setItem('category', category);
+        alert('Saved!')
+    }
     
     if (themeSwitcher.checked) {
         localStorage.setItem('theme', 'dark');
     } else {
         localStorage.setItem('theme', 'white');
     }
 
-    alert('Saved!')
-
 }
 
 function loadData() {
-    cardsField.innerHTML = '';
-    introImage.style.display = 'none';
-
-    let info = JSON.parse(localStorage.getItem('info'));
-    let category = localStorage.getItem('category');
+    let saved = localStorage.getItem('info');
 
-    generateCards(info, category);
+    if (saved !== "undefined") {
```

Comment:
Is it not enough - `if (saved) {....`  ?

Reply:
Yes, you are right, changed.